ca/config.go

```diff
@@ -138,27 +139,43 @@ func (s *SecurityConfig) UpdateRootCA(cert, key []byte, certExpiry time.Duration
 	}
 
 	// the RootCA pool should validate against the TLS certificate in the credentials
-	if s.ClientTLSCreds != nil {
-		s.ClientTLSCreds.UpdateCAs(rootCA.Pool, nil)
+	clientConfig := s.ClientTLSCreds.Config()
+	updatedClientConfig, err := NewClientTLSConfig(clientConfig.Certificates, rootCA.Pool, ManagerRole)
```

Comment:
I added a test which attempted to simultaneously update the certs and update the root CA.  It looks like I do need a lock on the root CA, because otherwise renewing the cert would overwrite the root CA changes.  :|